includes/class-events-store.php

```diff
@@ -882,6 +879,315 @@ public function count_events_by_status( $status ) {
+	 *
+	 * @param int   $event_id The ID of the event being updated.
+	 * @param array $row_data The row data used to update the event.
+	 * @return bool True if update was successful, false otherwise.
+	 */
+	public static function _update_event( int $event_id, array $row_data ): bool {
+		global $wpdb;
+
+		if ( empty( $event_id ) || empty( $row_data ) ) {
+			return 0;
+		}
+
+		$row_data['last_modified'] = current_time( 'mysql', true );
+
+		$where  = [ 'ID' => $event_id ];
+		$result = $wpdb->update( self::table_name(), $row_data, $where, self::row_formatting( $row_data ), self::row_formatting( $where ) );
+
+		self::flush_event_cache();
+		return false !== $result;
+	}
+
+	/**
+	 * Get raw event data by an ID.
+	 * For internal use only, please use Event::get( $id ).
+	 *
+	 * Currently no need for caching here really,
+	 * the action/instance/timestamp combination is the query that often happens on the FE.
+	 * So perhaps room for enhacement there later.
+	 *
+	 * @param int $id The ID of the event being retrieved.
+	 * @return object|null Raw event object if successful, false otherwise.
+	 */
+	public static function _get_event_raw( int $id ): ?object {
+		if ( $id <= 0 ) {
+			return null;
+		}
+
+		global $wpdb;
+		$table_name = self::table_name();
+
+		// Cannot prepare table name. @codingStandardsIgnoreLine
+		$row = $wpdb->get_row( $wpdb->prepare( "SELECT * FROM {$table_name} WHERE id = %d", $id ) );
+
+		return is_object( $row ) ? $row : null;
+	}
+
+	/**
+	 * Get raw events data based on various available query args.
+	 * For internal use only, please use Event::get( $args ) or Events::query( $args ).
+	 *
+	 * @param array $args Argument list for the query.
+	 * @return array Array of raw event objects.
+	 */
+	public static function _query_events_raw( array $args = [] ): array {
+		global $wpdb;
+
+		$valid_args = [
+			'action' => [
+				'default'    => null,
+				'validation' => 'is_string',
+			],
+			'action_hashed' => [
+				'default'    => null,
+				'validation' => 'is_string',
+			],
+			'args' => [
+				'default'    => null,
+				'validation' => 'is_array',
+			],
+			'instance' => [
+				'default'    => null,
+				'validation' => 'is_string',
+			],
+			'timestamp' => [
+				'default'    => null,
+				'validation' => fn( $ts ) => self::validate_timestamp( $ts ),
+			],
+			'schedule' => [
+				'default'    => null,
+				'validation' => 'is_string',
+			],
+			'status' => [
+				'default'    => self::ACTIVE_STATUSES,
+				'validation' => fn( $status ) => self::validate_status( $status ),
+			],
+			'limit' => [
+				'default'    => 1,
+				'validation' => 'is_int',
+			],
+			'page' => [
+				'default'    => 1,
+				'validation' => fn( $page ) => is_int( $page ) && $page >= 1,
+			],
+			'order' => [
+				'default'    => 'ASC',
+				'validation' => fn( $order ) => is_string( $order ) && in_array( strtoupper( $order ), [ 'ASC', 'DESC'], true ),
+			],
+		];
+
+		$parsed_args = wp_parse_args( $args, [
```

Comment:
This repeated list could be replaced with:
```
array_map(function( $argument ) { return $argument['default'] }, $valid_args );
```

Or even with:
```
array_map(function( $argument ) { return $argument['default']  ?? null }, $valid_args );
```
and then drop the lines setting `null` as defaults (most args) from above.

Reply:
Ah smart, I looked past the duplication here :). I'll go w/ the first I think and leave `null` defaults, as I kinda like how it helps document usage automatically in `$valid_args`.